directus-studio/index.md

```diff
@@ -0,0 +1,156 @@
+
+## Configuring Roles and Permissions
+
+Directus’s flexibility in permissions allows you to control access precisely:
+![Roles and permissions tab on Directus studio](access-control.png)
+
+1. **Navigate to Roles & Permissions** in Directus Studio and create two new roles named `Customer`, `Agent`. Agents will be assigned to review and treat tickets.
+2. **Set Permissions for the `Customer` Role:** Allow `read` and `create` access to the `support_tickets` collection. It's crucial to restrict `read` access to only their tickets, ensuring privacy and security.
+   ![Applying rules to what user can see](rule-setting.png)
+3. **Set Permissions for the `Agent` Role:** Allow `read` and `update` access to the `support_tickets` assigned to them. Also, grant a full `create`, `read`, and `update` role to the Agent on the `ticket_message` collection. 
+
+![Applying rules to Agent, to limit what they can see](directus-agent-role.gif)
+
+
+
+```
+:::Rule Setting
+
+Directus allows you to apply filters to permissions. To ensure users can only see their own tickets, you'll apply a filter on the Read permission. According to the image above, you'll allow the user to see only the tickets they created. You need to choose the field that links the ticket to the user. This is often a user ID or user email field in the ticket collection that references the user who submitted the ticket. This same goes to the permission set on the Agent role as well. You can read more about user fileds on the [documentation](https://docs.directus.io/reference/system/users.html).
+
+> Test this setting with a test account before going live.
+
+:::
+```
+
+## Inviting Customers to Use the Form
+
+To allow customers to submit tickets, you must add them to Directus and assign them the `Customer`/`User` role based on what the configuration is:
+
+1. Add Users in Directus Studio: Go to the Users section and invite customers via email.
+   ![Invite user to your directus platform](user-directory.png)
+2. **Assign Roles:** Ensure each customer user is assigned the `Customer` role to access and submit the support ticket form.
+
+## Basic System Walkthrough
+
+### Customer
+1. The customer logs into the Directus platform and navigates to the support ticket form.
+2. They fill out the form with their details and issue description and submit the ticket.
+3. The customer can then go to their ticket overview page to see the status of their ticket and any messages from support agents.
+
+### Support Agents
+1. The support agent views a dashboard of all tickets, filtered by their current status.
+2. Upon selecting a ticket, the agent can see all details provided by the customer and start a conversation through the messaging system within the ticket.
+3. As they work on resolving the ticket, the agent updates the ticket's status and, once resolved, adds a resolution summary for the customer to view.
```

Comment:
Screenshots for both please, as well as admin

Reply:
I'd also like to see the flow of adding a message, and setting up a flow to send email updates when there are new messages in a ticket.